### PR TITLE
Expose TLS prf

### PR DESCRIPTION
**Pull request by torben-hansen**
### Description of changes: 

`CRYPTO_tls1_prf` is validated and an exported symbol, but the declaration is private... Sort that out, such that it eventually will be easy to consume.

### Call-outs:

I put the declaration in `kdf.h`, since that seemed appropriate. Aimed for something that might work over several libcrypto types.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: include/openssl/kdf.h [hunk 1]
```diff
@@ -21,6 +21,14 @@
 extern "C" {
 #endif
 
+// tls1_prf calculates |out_len| bytes of the TLS PDF, using |digest|, and
```
Comment (andrewhop) [inline include/openssl/kdf.h:24]:
```suggestion
// CRYPTO_tls1_prf calculates |out_len| bytes of the TLS PDF, using |digest|, and
```

File: include/openssl/kdf.h [hunk 2]
```diff
@@ -83,7 +91,6 @@ OPENSSL_EXPORT int EVP_PKEY_CTX_add1_hkdf_info(EVP_PKEY_CTX *ctx,
                                                const uint8_t *info,
                                                size_t info_len);
 
-
```
Comment (andrewhop) [inline include/openssl/kdf.h:86]: You need 2 blank lines before the ending extern c++.

**Review by andrewhop — COMMENTED** [review]
Discussed offline if it's worth exposing this functionality through the OpenSSL [EVP_PKEY_TLS1_PRF](https://www.openssl.org/docs/man1.1.1/man3/EVP_PKEY_CTX_add1_tls1_prf_seed.html) API.

File: include/openssl/kdf.h (hunk 1 above)
Comment (dkostic) [inline include/openssl/kdf.h:24]:
```suggestion
// tls1_prf calculates |out_len| bytes of the TLS PRF, using |digest|, and
```

**Review by dkostic — APPROVED** [review]

File: include/openssl/kdf.h [hunk 3]
```diff
@@ -21,6 +21,16 @@
 extern "C" {
 #endif
 
+// CRYPTO_tls1_prf calculates |out_len| bytes of the TLS PRF, using |digest|,
+// and writes them to |out|. It returns one on success and zero on error.
+// TLS 1.2: https://datatracker.ietf.org/doc/html/rfc5246#section-5
+// TLS 1.{0,1}: https://datatracker.ietf.org/doc/html/rfc4346#section-5
+OPENSSL_EXPORT int CRYPTO_tls1_prf(const EVP_MD *digest,
```
Comment (andrewhop) [inline include/openssl/kdf.h:28]: Does this work for 1.3? Should the name/documentation be updated to make that clear? Also do we have a 1.3 specific function?

**Review by andrewhop — APPROVED** [review]

**Pull request merged**